tests/core/streams/Stream.spec.ts

```diff
@@ -0,0 +1,128 @@
+import Stream from 'src/core/streams/Stream';
+
+describe(`Stream`, () => {
+    it('can get the stream length', () => {
+        const bytes = Uint8Array.from([1, 1, 2]);
+        const stream = new Stream(bytes, 0, bytes.length);
+        expect(stream.length).toEqual(3);
+    });
+
+    it('can detect when the stream is empty', () => {
+        const stream = new Stream(Uint8Array.from([]), 0, 0);
+        expect(stream.isEmpty).toEqual(true);
+    });
+
+    describe('can get the bytes from a stream', () => {
+        const bytes = Uint8Array.from([1, 1, 2]);
+
+        it('when the current position is at or after the end', () => {
+            const stream = new Stream(bytes, 5);
+            expect(stream.getByte()).toEqual(-1);
+        });
+
+        it('when the current position is before the end', () => {
+            const stream = new Stream(bytes, 0);
+            for (let pos = 0; pos < bytes.length; pos++) {
+                expect(stream.getByte()).toEqual(bytes[pos]);
+            }
+        });
+    });
+
+    it('can get unsigned sixteen bit integers', () => {
+        const stream = new Stream(Uint8Array.from([1, 1, 2, 2]));
+        expect(stream.getUint16()).toEqual(257);
+        expect(stream.getUint16()).toEqual(514);
+    });
+
+    it('can get thirty-two bit integers', () => {
+        const bytes = Uint8Array.from([1, 1, 1, 1, 2, 2, 2, 2, 255, 255, 255, 255]);
+        const stream = new Stream(bytes);
+        expect(stream.getInt32()).toEqual(16843009);
+        expect(stream.getInt32()).toEqual(33686018);
+        expect(stream.getInt32()).toEqual(-1);
+    });
```

Comment:
This test confused me a bit at first. I initially thought the `-1` was returned because the end of the stream had been reached. But it's actually just because that's what `[255, 255, 255, 255]` evaluates to as a signed 32 bit integer. 

Can we swap around the order of the byte subsequences to eliminate the potential for confusion here? For example:
```js
it('can get thirty-two bit integers', () => {
  const bytes = Uint8Array.from([1, 1, 1, 1, 255, 255, 255, 255, 2, 2, 2, 2]);
  const stream = new Stream(bytes);
  expect(stream.getInt32()).toEqual(16843009);
  expect(stream.getInt32()).toEqual(-1);
  expect(stream.getInt32()).toEqual(33686018);
});
```

Reply:
Updated